### PR TITLE
filter the statvars on the menu based on places

### PR DESCRIPTION
**Pull request by qlj-lijuan**
- remove the checkbox when the statsvar is not available for selected place

### CODE REVIEW

File: static/js/tools/statsvar_menu.tsx [hunk 1]
```diff
@@ -3,17 +3,18 @@ import axios from "axios";
...
   c: number; // count
-  cd: NodePropType[];  // children
-  t: string;  // type
+  cd: NodePropType[]; // children
+  t: string; // type
   sv: string;
   updateUrl: (statvar: string, add: boolean) => void;
   nodePath: string;
   svPaths: string[][];
+  svAvai: string[];
```
Comment (shifucun) [inline static/js/tools/statsvar_menu.tsx:17]: maybe rename to svValid? "Avai" is not that easy to guess the meaning

File: static/js/tools/statsvar_menu.tsx [hunk 2]
```diff
@@ -95,21 +97,22 @@ class Node extends Component<NodePropType, NodeStateType> {
...
-                {this.props.c !== 0 && "(" + this.props.c + ")"}
-              </sup>
-              {checkboxImg}
-              {expandImg}
-            </a>
-          </span>
-          {child}
-        </li>
-      </ul>
+      (typeof checkboxImg !== "undefined" ||
```
Comment (shifucun) [inline static/js/tools/statsvar_menu.tsx:100]: better to add a boolean class variable to record this than check the undefined. Can set the boolean around line 81

File: static/js/tools/timeline_page.tsx [hunk 3]
```diff
@@ -81,16 +86,19 @@ class Page extends Component<PagePropType, PageStateType> {
       }
     }
 
-    Promise.all([statvarInfoPromise, placesPromise]).then((values) => {
-      for(let idx=0; idx<svIds.length; idx++){
-        values[0][svIds[idx]].title=svPaths[idx][svPaths[idx].length -1]
+    Promise.all([statvarInfoPromise, placesPromise, statvarAvaiPromise]).then(
+      (values) => {
+        for (let idx = 0; idx < svIds.length; idx++) {
+          values[0][svIds[idx]].title = svPaths[idx][svPaths[idx].length - 1];
```
Comment (shifucun) [inline static/js/tools/timeline_page.tsx:92]: use .slice()[-1] for better readability

File: static/js/tools/timeline_util.js [hunk 4]
```diff
@@ -170,6 +170,28 @@ function getStatsVarInfo(dcids) {
   });
 }
 
+function getStatsVar(dcids) {
```
Comment (shifucun) [inline static/js/tools/timeline_util.js:173]: can you make this file a typescript file and add typing.

File: static/js/tools/statsvar_menu.tsx (hunk 1 above)
Comment (qlj-lijuan) [reply static/js/tools/statsvar_menu.tsx]: done.

File: static/js/tools/statsvar_menu.tsx (hunk 2 above)
Comment (qlj-lijuan) [reply static/js/tools/statsvar_menu.tsx]: done.

File: static/js/tools/timeline_page.tsx (hunk 3 above)
Comment (qlj-lijuan) [reply static/js/tools/timeline_page.tsx]: done.

File: static/js/tools/timeline_util.js (hunk 4 above)
Comment (qlj-lijuan) [reply static/js/tools/timeline_util.js]: done.

File: static/js/tools/timeline_page.tsx [hunk 5]
```diff
@@ -70,8 +73,10 @@ class Page extends Component<PagePropType, PageStateType> {
     }
 
     let placesPromise = Promise.resolve(this.state.places);
+    let statvarValidPromise = Promise.resolve(this.state.statvarValid);
```
Comment (shifucun) [inline static/js/tools/timeline_page.tsx:76]: nit: validStatsVarPromise

File: static/js/tools/timeline_util.ts [hunk 6]
```diff
@@ -170,6 +176,28 @@ function getStatsVarInfo(dcids) {
...
+  const promises = [];
+  for (const dcid of dcids) {
+    promises.push(
+      axios.get("/api/place/statsvars/" + dcid).then((resp) => {
+        return resp.data;
+      })
+    );
+  }
+  return Promise.all(promises).then((values) => {
+    let statvars = ["Count_Person"]; // Count_Person not in List ???
```
Comment (shifucun) [inline static/js/tools/timeline_util.ts:192]: can use javascript set here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set, should be faster with large number items

File: static/js/tools/timeline_util.ts [hunk 7]
```diff
@@ -170,6 +176,28 @@ function getStatsVarInfo(dcids) {
...
+    return Promise.resolve([]);
+  }
+  const promises = [];
+  for (const dcid of dcids) {
+    promises.push(
+      axios.get("/api/place/statsvars/" + dcid).then((resp) => {
+        return resp.data;
+      })
+    );
+  }
```
Comment (shifucun) [inline static/js/tools/timeline_util.ts:190]: can you add a todo here to read the set from server side

File: static/js/tools/timeline_page.tsx (hunk 5 above)
Comment (qlj-lijuan) [reply static/js/tools/timeline_page.tsx]: done.

File: static/js/tools/timeline_util.ts (hunk 7 above)
Comment (qlj-lijuan) [reply static/js/tools/timeline_util.ts]: done.

File: static/js/tools/timeline_util.ts (hunk 6 above)
Comment (qlj-lijuan) [reply static/js/tools/timeline_util.ts]: done.

File: static/js/tools/timeline_util.ts [hunk 8]
```diff
@@ -15,20 +15,50 @@
  */
 import axios from "axios";
 import { getUrlVars, setSearchParam } from "./dc";
-import { SEP } from "./statsvar_menu.tsx";
+import { SEP } from "./statsvar_menu";
+import { keys } from "../../webpack.config";
```
Comment (shifucun) [inline static/js/tools/timeline_util.ts:19]: what is this keys used for?

Comment (qlj-lijuan) [reply static/js/tools/timeline_util.ts]: Oops. I think it's added by mistake. I removed this line.

**Pull request merged**